addons/purchase/models/purchase.py

```diff
@@ -3,10 +3,11 @@
 
 from datetime import datetime
 from dateutil.relativedelta import relativedelta
+from itertools import groupby
 
 from odoo import api, fields, models, SUPERUSER_ID, _
 from odoo.osv import expression
-from odoo.tools import DEFAULT_SERVER_DATETIME_FORMAT
+from odoo.tools import DEFAULT_SERVER_DATETIME_FORMAT, float_is_zero, float_compare
```

Comment:
float_compare is already imported bellow from odoo.tools.float_utils import float_compare. Also add float_is_zero on the float_utils line